express/components/hf_datasets_components.py

```diff
@@ -0,0 +1,131 @@
+"""Hugging Face Datasets single component module """
+
+import os
+import importlib
+import tempfile
+from abc import ABC, abstractmethod
+from typing import List, Optional, Dict, Union
+
+import datasets
+from datasets import load_dataset
+
+from .common import ExpressDatasetHandler, ExpressDataset, ExpressTransformComponent, \
+    ExpressDatasetDraft, ExpressLoaderComponent
+from express.storage_interface import StorageHandlerModule
+from express.manifest import DataManifest, DataSource, DataType
+
+# Define interface of pandas draft
+# pylint: disable=unsubscriptable-object
+HFDatasetsDatasetDraft = ExpressDatasetDraft[List[str], Union[pd.DataFrame, pd.Series]]
+
+# pylint: disable=no-member
+STORAGE_MODULE_PATH = StorageHandlerModule().to_dict()[os.environ.get('CLOUD_ENV', 'GCP')]
+STORAGE_HANDLER = importlib.import_module(STORAGE_MODULE_PATH).StorageHandler()
+
+
+# pylint: disable=too-few-public-methods
+class HFDatasetsDataset(ExpressDataset[List[str], Union[datasets.Dataset, datasets.DatasetDict]]):
```

Comment:
Also was wondering whether we could have a better name (DatasetsDataset isn't the best name)